modules/core/src/lib/editable-feature-collection.js

```diff
@@ -121,11 +123,59 @@ export class EditableFeatureCollection {
         handles = handles.concat(getEditHandlesForGeometry(this._tentativeFeature.geometry, -1));
         // Slice off the handles that are are next to the pointer
         if (this._tentativeFeature && this._tentativeFeature.geometry.type === 'LineString') {
-          // Remove the last existing and intermediate handles
-          handles = handles.slice(0, -2);
+          // Remove the last existing handle
+          handles = handles.slice(0, -1);
         } else if (this._tentativeFeature && this._tentativeFeature.geometry.type === 'Polygon') {
-          // Remove the last existing and 2 intermediate handles
-          handles = handles.slice(0, -3);
+          // Remove the last existing handle
+          handles = handles.slice(0, -1);
+        }
+      }
+    }
+
+    // intermediate edit handle
+    if (picks && picks.length && groundCoords) {
+      const existingEditHandle = picks.find(
+        pick => pick.isEditingHandle && pick.object && pick.object.type === 'existing'
+      );
+      // don't show intermediate point when too close to an existing edit handle
+      const featureAsPick = !existingEditHandle && picks.find(pick => !pick.isEditingHandle);
+
+      // is the feature in the pick selected
+      if (featureAsPick && this._selectedFeatureIndexes.includes(featureAsPick.index)) {
```

Comment:
Does this also need to check geometry type has lines? Something like `if (featureAsPick.geometry.type !== 'MultiPoint' && featureAsPick.geometry.type !== 'Point)`?